lib/index.ts

```diff
@@ -54,11 +56,42 @@ export function getCommand(root: string, targetFile: string | undefined) {
   return 'mvn';
 }
 
+function getParentDirectory(p: string): string {
+  return path.dirname(p);
+}
+
 // When we have `mvn`, we can run the subProcess from anywhere.
 // However due to https://github.com/takari/maven-wrapper/issues/133, `mvnw` can only be run
 // within the directory where `mvnw` exists
-function calculateTargetFilePath(mavenCommand, root: string, targetPath) {
-  return mavenCommand === 'mvn' ? root : path.dirname(targetPath);
+function resolveExecutionDirectory(mavenCommand, root: string, targetPath) {
```

Comment:
looks good! might be nice to keep it similar structure to what happens in Gradle plugin (findWrapper() etc) just for consistency

Reply:
I think I will move `gradle` to this structure, right now it looks like this:
```ts
  const command = getCommand(root, targetFile);
...
  try {
    gradleVersionOutput = await subProcess.execute(command, ['-v'], {
      cwd: root,
....